src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-list/osd-list.component.ts

```diff
@@ -121,4 +220,82 @@ export class OsdListComponent implements OnInit {
   configureClusterAction() {
     this.bsModalRef = this.modalService.show(OsdFlagsModalComponent, {});
   }
+
+  showConfirmationModal(markAction: string, onSubmit: (id: number) => Observable<any>) {
+    this.bsModalRef = this.modalService.show(ConfirmationModalComponent, {
+      initialState: {
+        titleText: `Mark OSD ${markAction}`,
+        buttonText: `Mark ${markAction}`,
+        bodyTpl: this.markOsdConfirmationTpl,
+        bodyContext: {
+          markActionDescription: markAction
+        },
+        onSubmit: () => {
+          onSubmit
+            .call(this.osdService, this.selection.first().id)
+            .subscribe(() => this.bsModalRef.hide());
+        }
+      }
+    });
+  }
+
+  markOut() {
+    this.showConfirmationModal('out', this.osdService.markOut);
+  }
+
+  markIn() {
+    this.showConfirmationModal('in', this.osdService.markIn);
+  }
+
+  markDown() {
+    this.showConfirmationModal('down', this.osdService.markDown);
+  }
+
+  reweight() {
+    const selectedOsd = this.osds.filter((o) => o.id === this.selection.first().id).pop();
+    this.modalService.show(OsdReweightModalComponent, {
+      initialState: {
+        currentWeight: selectedOsd.weight,
+        osdId: selectedOsd.id
+      }
+    });
+  }
+
+  showCriticalConfirmationModal(
```

Comment:
Could you please rename it?
```suggestion
  showCriticalActionModal(
```